include/ia2.h

```diff
@@ -26,6 +26,11 @@
 #define IA2_WRPKRU "wrpkru"
 #endif
 
+// The callback passed to dl_iterate_phdr in the constructor inserted by
+// INIT_COMPARTMENT to pkey_mprotect the pages corresponding to the
+// compartment's loaded segments.
+int protect_pages(struct dl_phdr_info *info, size_t size, void *data);
```

Comment:
Maybe we should add an `#ifndef _GNU_SOURCE/#error "This header requires using -D_GNU_SOURCE"` since we need link.h to declare `dl_phdr_info` and it's not always clear what the issue is when it's not defined.

Reply:
Good idea! Done.